indexd/src/app_client.rs

```diff
@@ -238,8 +254,25 @@ impl Client {
     }
 
     /// Returns all usable hosts.
-    pub async fn hosts(&self) -> Result<Vec<Host>> {
-        self.get_json::<_, ()>("hosts", None).await
+    ///
+    /// # Arguments
+    /// * `query` - Optional parameters to control the hosts listing.
+    pub async fn hosts(&self, query: Option<HostQuery>) -> Result<Vec<Host>> {
```

Comment:
I'd drop the `Option` and instead implement `Default` for `HostQuery` which returns a `HostQuery` struct. The default can then have `None` for `sort`. 

Then you can call `hosts` with `default()` and people who want to construct a custom `HostQuery` can use `default()` to only set the fields they want and keep defaults for the remaining ones. That also means that adding more fields to `HostQuery` won't break existing calls of `hosts` since it will just fill in defaults.